Userland/Libraries/LibC/malloc.cpp

```diff
@@ -153,12 +154,64 @@ static inline BigAllocator (&big_allocators())[1]
     return reinterpret_cast<BigAllocator(&)[1]>(g_big_allocators_storage);
 }
 
-static Allocator* allocator_for_size(size_t size, size_t& good_size)
+// --- BEGIN MATH ---
+// This stuff is only used for checking if there exists an aligned block in a
+// chunk. It has no bearing on the rest of the allocator, especially for
+// regular malloc.
+
+static inline unsigned long modulo(long a, long b)
+{
+    return (b + (a % b)) % b;
+}
+
+struct EuclideanResult {
+    long x;
+    long y;
+    long gcd;
+};
+
+// Returns x, y, gcd.
+static inline EuclideanResult extended_euclid(long a, long b)
+{
+    if (b == 0)
+        return { 1, 0, a };
+
+    auto [x, y, gcd] = extended_euclid(b, a % b);
```

Comment:
I don't think the extended euclid algorithm needs recursion to work - maybe turn this into a loop?

Reply:
done

Reply:
Nice!